.ci/test.py

```diff
@@ -306,20 +311,82 @@ def run_all(args):
     root = Path(root_path)
 
     plugins = list(enumerate_plugins(root))
-    if args != []:
-        plugins = [p for p in plugins if p.name in args]
+    if plugin_names != []:
+        plugins = [p for p in plugins if p.name in plugin_names]
         print("Testing the following plugins: {names}".format(names=[p.name for p in plugins]))
     else:
         print("Testing all plugins in {root}".format(root=root))
 
     results = [(p, run_one(p)) for p in plugins]
     success = all([t[1] for t in results])
 
+    if sys.version_info[0:2] == (3, 12) and update_badges:
+        push_badges_data(collect_badges_data(results, success), workflow)
+
     if not success:
         print("The following tests failed:")
         for t in filter(lambda t: not t[1], results):
             print(" - {p.name} ({p.path})".format(p=t[0]))
         sys.exit(1)
+    else:
+        print("All tests passed.")
+
+
+def collect_badges_data(results, success):
+    badges_data = {}
+    for t in results:
+        p = t[0]
+        if has_testfiles(p):
+            if success or t[1]:
+                badges_data[p.name] = True
+            else:
+                badges_data[p.name] = False
+    return badges_data
+
+
+def configure_git():
+    subprocess.run(["git", "config", "--global", "user.email", '"lightningd@plugins.repo"'])
+    subprocess.run(["git", "config", "--global", "user.name", '"lightningd"'])
```

Comment:
Clarify that this is an automated / bot user

Reply:
Added a comment: https://github.com/lightningd/plugins/pull/509/commits/a4cc954d38bac9002d508b8e17e14a307869170f